client/trino-client/src/main/java/io/trino/client/uri/AbstractConnectionProperty.java

```diff
@@ -82,78 +83,81 @@ public PropertyName getPropertyName()
         return propertyName;
     }
 
-    @Override
-    public DriverPropertyInfo getDriverPropertyInfo(Properties mergedProperties)
-    {
-        String currentValue = mergedProperties.getProperty(key);
-        DriverPropertyInfo result = new DriverPropertyInfo(key, currentValue);
-        result.required = isRequired.test(mergedProperties);
-        result.choices = (choices != null) ? choices.clone() : null;
-        return result;
-    }
-
     @Override
     public boolean isRequired(Properties properties)
     {
         return isRequired.test(properties);
     }
 
-    @Override
-    public boolean isValid(Properties properties)
-    {
-        return !validator.validate(properties).isPresent();
-    }
-
     @Override
     public Optional<T> getValue(Properties properties)
-            throws SQLException
-    {
-        return getValueOrDefault(properties, defaultValue);
-    }
-
-    @Override
-    public Optional<T> getValueOrDefault(Properties properties, Optional<T> defaultValue)
-            throws SQLException
     {
-        V value = (V) properties.get(key);
+        Object value = properties.get(key);
         if (value == null) {
             if (isRequired(properties) && !defaultValue.isPresent()) {
-                throw new SQLException(format("Connection property %s is required", key));
+                throw new RuntimeException(format("Connection property %s is required", key));
```

Comment:
Connection properties are shared between the CLI and client and JDBC. Exception is wrapped with SQLException in the Driver